### PR TITLE
allow Batch extraopts to be changed at resubmission

### PR DESCRIPTION
**Pull request by ryuwd**
Try to allow `extraopts` to be changed at resubmit, which is helpful for resubmitting subjobs, etc to Batch systems with modified runtime or memory resource parameters.

I have marked this as draft as I'm not sure how `Schema` works. Am I right to increment the minor version?

### CODE REVIEW

**Comment by ryuwd** [comment]
I tried this out but `Batch` resubmit method doesn't take a `backend` kwarg, so resubmitting doesn't work.

**Comment by egede** [comment]
The Schema options are in effect never used, so they can actually just be ignored (some clean-up required).

**Comment by ryuwd** [comment]
Adding `changable_at_resubmit=1` is enough to let me modify the `extraopts` attribute on the existing backend and let me `resubmit()` without any problems.

(Before I was trying to pass a new backend to `resubmit`, which doesn't seem to be supported for most backends anyway.)

**Review by mesmith75 — APPROVED** [review]

File: ganga/GangaCore/Lib/Batch/Batch.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ class Batch(IBackend):
     backend.extraopts = '-t 07:14:12:59' #Killed if not finished by 14 July before 1 pm
     """
     _schema = Schema(Version(1, 0), {'queue': SimpleItem(defvalue='', doc='queue name as defomed in your local Batch installation'),
-                                     'extraopts': SimpleItem(defvalue='', doc='extra options for Batch. See help(Batch) for more details'),
+                                     'extraopts': SimpleItem(defvalue='', changable_at_resubmit=1, doc='extra options for Batch. See help(Batch) for more details'),
```
Comment (mesmith75) [inline ganga/GangaCore/Lib/Batch/Batch.py:94]: Could you check the spelling?

Comment (mesmith75) [reply ganga/GangaCore/Lib/Batch/Batch.py:94]: Ah, its right.

**Comment by ryuwd** [comment]
Thanks @mesmith75!

**Pull request merged**